src/OpenTelemetry.Instrumentation.StackExchangeRedis/Implementation/RedisProfilerEntryToActivityConverter.cs

```diff
@@ -120,28 +124,41 @@ static bool GetCommandAndKey(
             // Total:
             // command.ElapsedTime;             // 00:00:32.4988020
 
-            if (options.SetVerboseDatabaseStatements)
+            if (options.EmitOldAttributes)
             {
-                var (commandAndKey, script) = MessageDataGetter.Value.Invoke(command);
+                activity.SetTag(StackExchangeRedisConnectionInstrumentation.RedisDatabaseIndexKeyName, command.Db);
 
-                if (!string.IsNullOrEmpty(commandAndKey) && !string.IsNullOrEmpty(script))
-                {
-                    activity.SetTag(SemanticConventions.AttributeDbStatement, commandAndKey + " " + script);
-                }
-                else if (!string.IsNullOrEmpty(commandAndKey))
+                if (options.SetVerboseDatabaseStatements)
                 {
-                    activity.SetTag(SemanticConventions.AttributeDbStatement, commandAndKey);
+                    var (commandAndKey, script) = MessageDataGetter.Value.Invoke(command);
+
+                    if (!string.IsNullOrEmpty(commandAndKey) && !string.IsNullOrEmpty(script))
+                    {
+                        activity.SetTag(SemanticConventions.AttributeDbStatement, commandAndKey + " " + script);
+                    }
+                    else if (!string.IsNullOrEmpty(commandAndKey))
+                    {
+                        activity.SetTag(SemanticConventions.AttributeDbStatement, commandAndKey);
+                    }
+                    else if (command.Command != null)
+                    {
+                        // Example: "db.statement": SET;
+                        activity.SetTag(SemanticConventions.AttributeDbStatement, command.Command);
+                    }
                 }
                 else if (command.Command != null)
                 {
                     // Example: "db.statement": SET;
                     activity.SetTag(SemanticConventions.AttributeDbStatement, command.Command);
                 }
```

Comment:
Maybe refactor this to something like this to avoid the repetition/duplication?

```csharp
string? statement = null;

if (options.SetVerboseDatabaseStatements)
{
    var (commandAndKey, script) = MessageDataGetter.Value.Invoke(command);

    if (!string.IsNullOrEmpty(commandAndKey))
    {
        statement = commandAndKey;

        if (!string.IsNullOrEmpty(script))
        {
            statement += " " + script;
        }
    }
}

// Example: "db.statement": SET;
statement ??= command.Command;

if (statement != null)
{
    activity.SetTag(SemanticConventions.AttributeDbStatement, statement);
}
```